code/modules/admin/verbs/adminevents.dm

```diff
@@ -34,33 +34,42 @@
 	admin_headset_message(M)
 
 /client/proc/admin_headset_message(mob/M in GLOB.mob_list, sender = null)
-	var/mob/living/carbon/human/H = M
+	var/mob/living/carbon/human/human_recipient
+	var/mob/living/silicon/silicon_recipient
 
 	if(!check_rights(R_ADMIN))
 		return
 
-	if(!istype(H))
-		to_chat(usr, "This can only be used on instances of type /mob/living/carbon/human", confidential = TRUE)
-		return
-	if(!istype(H.ears, /obj/item/radio/headset))
-		to_chat(usr, "The person you are trying to contact is not wearing a headset.", confidential = TRUE)
+
+	if(ishuman(M))
+		human_recipient = M
+		if(!istype(human_recipient.ears, /obj/item/radio/headset))
+			to_chat(usr, "The person you are trying to contact is not wearing a headset.", confidential = TRUE)
+			return
+	else if(issilicon(M))
+		silicon_recipient = M
+		if(!istype(silicon_recipient.radio, /obj/item/radio))
+			to_chat(usr, "The silicon you are trying to contact does not have a radio installed.", confidential = TRUE)
+			return
+	else
+		to_chat(usr, "This can only be used on instances of type /mob/living/carbon/human or /mob/living/silicon", confidential = TRUE)
 		return
 
 	if (!sender)
 		sender = input("Who is the message from?", "Sender") as null|anything in list(RADIO_CHANNEL_CENTCOM,RADIO_CHANNEL_SYNDICATE)
 		if(!sender)
 			return
 
-	message_admins("[key_name_admin(src)] has started answering [key_name_admin(H)]'s [sender] request.")
-	var/input = input("Please enter a message to reply to [key_name(H)] via their headset.","Outgoing message from [sender]", "") as text|null
+	message_admins("[key_name_admin(src)] has started answering [key_name_admin(human_recipient ? human_recipient : silicon_recipient)]'s [sender] request.")
+	var/input = input("Please enter a message to reply to [key_name(human_recipient ? human_recipient : silicon_recipient)] via their headset.","Outgoing message from [sender]", "") as text|null
```

Comment:
there's no need to keep doing the `?:` operator check, just cache it to a var beforehand like

```txt
var/target = human_recipient || silicon_recipient
```

and then use that instead in all of these procs